frontend/app/src/pages/main/workers/managed-workers/$managed-worker/components/managed-worker-activity.tsx

```diff
@@ -0,0 +1,194 @@
+import { queries } from '@/lib/api';
+import { useQuery } from '@tanstack/react-query';
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from '@/components/ui/card';
+import { Spinner } from '@/components/ui/loading';
+import RelativeDate from '@/components/molecules/relative-date';
+import { Button } from '@/components/ui/button';
+import { ArrowRightIcon, ChevronLeftIcon } from '@radix-ui/react-icons';
+import { cn } from '@/lib/utils';
+import { useState } from 'react';
+import {
+  ManagedWorker,
+  ManagedWorkerEvent,
+  ManagedWorkerEventStatus,
+} from '@/lib/api/generated/cloud/data-contracts';
+import { Separator } from '@/components/ui/separator';
+import { ManagedWorkerBuild } from './managed-worker-build';
+import GithubButton from './github-button';
+
+export function ManagedWorkerActivity({
+  managedWorker,
+}: {
+  managedWorker: ManagedWorker | undefined;
+}) {
+  const [buildId, setBuildId] = useState<string | undefined>();
+
+  if (buildId) {
+    return <Build buildId={buildId} back={() => setBuildId(undefined)} />;
+  }
+
+  return <EventList managedWorker={managedWorker} setBuildId={setBuildId} />;
+}
+
+function EventList({
+  managedWorker,
+  setBuildId,
+}: {
+  managedWorker: ManagedWorker | undefined;
+  setBuildId: (id: string) => void;
+}) {
+  const getLogsQuery = useQuery({
+    ...queries.cloud.listManagedWorkerEvents(managedWorker?.metadata.id || ''),
```

Comment:
```suggestion
    ...queries.cloud.listManagedWorkerEvents(managedWorker!.metadata.id),
```
enabled should cover unknown?